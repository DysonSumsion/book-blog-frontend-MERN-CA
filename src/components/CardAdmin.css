.cardAdmin {
  background-color: rgba(126, 211, 196, 0.5);
  width: 100vw;
  margin-bottom: 5px;
  display: flex;
  padding: 50px;
}

.cardAdminLeft {
  width: 150px;
}

.cardAdminLeftTop {
  height: 200px;
  /* background-color: pink; */
}

.cardAdminLeftBottom {
  /* background-color: yellow; */
  height: 100px;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
}

.cardAdminRight {
  padding-left: 20px;
  max-width: 700px;
}

img {
  max-width: 150px;
  object-fit: contain;
}
