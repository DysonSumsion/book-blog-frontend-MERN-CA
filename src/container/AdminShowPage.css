.showreviews {
  width: 100vw;
}