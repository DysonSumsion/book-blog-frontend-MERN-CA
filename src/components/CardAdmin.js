import React from "react";
import './CardAdmin.css';

function CardAdmin(props) {

  return (
    <div className="cardAdmin">

      <div className="cardAdminLeft">
        <div className="cardAdminLeftTop">
          <img src={props.url} alt={props.title}  ></img>
          </div>
        <div className="cardAdminLeftBottom">
          <button onClick={() => props.handleEditClick(props.review)} className="edit"> Edit</button>
          <button onClick={() => props.handleDeleteClick(props.title)} className="delete"> Delete</button>
        </div>
      </div>

      <div className="cardAdminRight">
        <p>Title: {props.title}</p>
        <p>Author: {props.author}</p>
        <p>Review: {props.review}</p>
        <p>Publisher: {props.publisher}</p>
        <p>Year Published: {props.yearPublished}</p>
        <p>Genre: {props.genre}</p>
        <p>ISBN: {props.isbn}</p>
        <p>URL to buy: {props.url}</p>
        <p>SEO Keywords: {props.keywords}</p>
      </div>

    </div>
  );
}

export default CardAdmin;