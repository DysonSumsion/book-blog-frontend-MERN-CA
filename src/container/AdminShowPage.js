import React from "react";
import IntroSection from "../components/IntroSection";
import CardAdmin from "../components/CardAdmin";
import axios from "axios";
import FormAdd from "../container/FormAdd";
import './AdminShowPage.css';
import {Link} from 'react-router-dom';

class AdminShowPage extends React.Component {
  state = {
    data: "",
    adding: false,
    selectedReview: null,
    id: ""
  }

  async componentDidMount() {

    try {
    const token = localStorage.getItem("token");
    const headers = {token: token}
    const response1 = await axios.get(`${process.env.REACT_APP_API_URL}/protected/reviews`, {headers})
    const data1 = response1.data
    
    // const response = await fetch(`${process.env.REACT_APP_API_URL}/reviews`)
    // console.log(response);
    // const data = await response.json()

    this.setState({
      data: data1
    })
    } catch(err) {
      console.log(err)
    }
    }

  renderReviews = reviewList => {
    console.log("in render reviews");
    // console.log(reviewList);
    return reviewList.map((review, index) => {
      return (
        < div className="reviewList" key={index}>
          <CardAdmin 
          title={review.title}
          author={review.author.name}
          review={review.review}
          publisher={review.publisher.name}
          yearPublished={review.yearPublished}
          isbn={review.isbn}
          linkToBuy={review.linkToBuy}
          genre={review.genre.name}
          topPick={review.topPick}
          url={review.url}
          handleDeleteClick={this.handleDeleteClick}
          handleEditClick={this.handleEditClick}
          reviewObject={review}
          />
        </div>
      )
    })
  }

  handleEditClick=(review) => {
    console.log(review);
    const id = review._id
    this.setState({ adding: true, selectedReview: review, id: id })
  }

  refresh = (reviewsData)  => {
    console.log(reviewsData);
    const reviews = reviewsData.data
    console.log(reviews);
    this.setState({
      data: reviews,
      adding: false
    })
  }

  handleDeleteClick= async (title) => { 
    const data = {data: {title}}
    const token = localStorage.getItem("token");
    console.log(token);
    const headers = {token: token}
    // const headers = {headers: {token: token}}

    await axios.delete(`${process.env.REACT_APP_API_URL}/protected/deleteReview`, {data, headers})
          .then((res)=>{
            console.log(res);
            alert("Review Deleted")
            window.location.reload()
          })
          .catch((err) => {
            console.log(err.response)
          })
  }

  deleteToken= (e) => {
    // e.preventDefault();
    window.localStorage.removeItem("token");
  }

  render() {
    // console.log(this.state);
  const { reviews } = this.state.data
  if (!reviews) {
    return <h2>Loading.......</h2>
  } 
  const result = this.renderReviews(reviews)

    return (
      <>
        <div>

          <IntroSection headingOne="Hi Jaclyn!" 
          headingTwo="Welcome to your dashboard, here are all of your reviews..." />

          {/* {result} */}
          <Link to = '/adminjaclyn' onClick={this.deleteToken}> Logout </Link>
          <Link to = '/auth/adminaddreview'> add review </Link>

          <div className="showreviews">
          {this.state.adding ? 
          <FormAdd
            title={this.state.selectedReview.title} 
            author={this.state.selectedReview.author.name}
            review={this.state.selectedReview.review}
            publisher={this.state.selectedReview.publisher.name}
            yearPublished={this.state.selectedReview.yearPublished}
            isbn={this.state.selectedReview.isbn}
            linkToBuy={this.state.selectedReview.linkToBuy}
            genre={this.state.selectedReview.genre.name}
            topPick={this.state.selectedReview.topPick}
            seoKeyword={this.state.selectedReview.seoKeyword}
            url={this.state.selectedReview.url}
            id={this.state.id}
            refresh={this.refresh}
            adding={this.state.adding}
            /> 
            : <div>{result}</div>
            }
          </div>
        </div>
      </>
    );
  }
}

export default AdminShowPage;
