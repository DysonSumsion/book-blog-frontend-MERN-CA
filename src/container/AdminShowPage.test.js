import React from "react";
import Enzyme, { shallow, render, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { createSerializer } from "enzyme-to-json";
import sinon from "sinon";
// // import FormAdd from './FormAdd'
import AdminShowPage from "./AdminShowPage";
import CardAdmin from '../components/CardAdmin'


describe("<AdminShowPage />", () => {
  const reviewList = [ 
    {
    title: "Three Litte Pigs",
    author: "Big Bad Wolf",
    _id: "5d19cd74a0c3e2c7f68ff800",
    url: "https://bookmarks-rag.s3.amazonaws.com/bookmarks-content.jpeg"
    },
    {
      title: "Go Blues",
      author: "Cheer Squad",
      _id: "5d19cd74a0c3e2c7f68ff801",
      url: "https://bookmarks-rag.s3.amazonaws.com/bookmarks-content.jpeg"
    },
    {
      title: "Tim Tams are Awesome",
      author: "Robert De Matteo",
      _id: "5d19cd74a0c3e2c7f68ff802",
      url: "https://bookmarks-rag.s3.amazonaws.com/bookmarks-content.jpeg"
    }
  ]



  it("displays loading when no review in list", () => {
    const review = [];
    const wrapper = shallow(
        <AdminShowPage data={review}  />
    );
    expect(wrapper).toMatchSnapshot();
    });

  it("renders CardAdmin component ", () => {
    const wrapper = shallow(
        <CardAdmin  />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("renders .cardLeft ", () => {
    const wrapper = shallow(
        <CardAdmin  />
    );
    expect(wrapper.find('.cardLeft')).toMatchSnapshot();
  });






  it("displays title in title field ", () => {

    const wrapper = render(
        <CardAdmin title={reviewList[0].title} renderReviews={() => {}} />
    );
    expect(wrapper).toMatchSnapshot();
    });

  it("displays author in author field ", () => {
    const wrapper = mount(
        <CardAdmin author={reviewList[0].author} renderReviews={() => {}} />
    );
    expect(wrapper).toMatchSnapshot();
    });




});


