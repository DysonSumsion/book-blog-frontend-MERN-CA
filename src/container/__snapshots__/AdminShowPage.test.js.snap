// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminShowPage /> displays author in author field  1`] = `
<div
  className="card"
>
  <div
    className="cardLeft"
  >
    <img />
  </div>
  <div
    className="cardRight"
  >
    <p>
      Title: 
    </p>
    <p>
      Author: 
      Big Bad Wolf
    </p>
    <button
      className="edit"
      onClick={[Function]}
    >
       Edit
    </button>
    <button
      className="delete"
      onClick={[Function]}
    >
       Delete
    </button>
  </div>
</div>
`;

exports[`<AdminShowPage /> displays loading when no review in list 1`] = `
<h2>
  Loading.......
</h2>
`;

exports[`<AdminShowPage /> displays title in title field  1`] = `
<div
  class="card"
>
  <div
    class="cardLeft"
  >
    <img
      alt="Three Litte Pigs"
    />
  </div>
  <div
    class="cardRight"
  >
    <p>
      Title: Three Litte Pigs
    </p>
    <p>
      Author: 
    </p>
    <button
      class="edit"
    >
       Edit
    </button>
    <button
      class="delete"
    >
       Delete
    </button>
  </div>
</div>
`;

exports[`<AdminShowPage /> renders .cardLeft  1`] = `
<div
  className="cardLeft"
>
  <img />
</div>
`;

exports[`<AdminShowPage /> renders CardAdmin component  1`] = `
<div
  className="card"
>
  <div
    className="cardLeft"
  >
    <img />
  </div>
  <div
    className="cardRight"
  >
    <p>
      Title: 
    </p>
    <p>
      Author: 
    </p>
    <button
      className="edit"
      onClick={[Function]}
    >
       Edit
    </button>
    <button
      className="delete"
      onClick={[Function]}
    >
       Delete
    </button>
  </div>
</div>
`;
